File: 2022/python/01/challenge_one.py
with open('data.txt', 'r') as file:
    elf= [0]
    elf_num: int = 0
    elf_cal = 0
    for line in file:
        if line == "\n":
            elf[elf_num] = elf_cal
            elf_cal = 0
            elf.append(0)
            elf_num += 1
        else:
            elf_cal += int(line)
    
    print(f"The higest cal carying by one Elf is {max(elf)}")
        

File: 2022/lua/01/challenge_one.lua
file_name = "data.txt"
file = io.open(file_name, "rb")

if file then 
    file:close()
else
    print("File not found") 
    return 
end

elves = {}
elves_counter = 0
cal = 0
for line in io.lines(file_name) do 
    if line == "" then 
        elves[elves_counter] = cal
        elves_counter = elves_counter+1
        cal = 0 
    else
        cal = cal + tonumber(line)
    end
end

print(math.max(table.unpack(elves)))
